sensitivity-analysis-api/src/main/java/com/powsybl/sensitivity/SensitivityFactor.java

```diff
@@ -57,12 +57,12 @@ public class SensitivityFactor {
      * @param contingencyContext see {@link com.powsybl.contingency.ContingencyContext}
      */
     public SensitivityFactor(SensitivityFunctionType functionType, String functionId, SensitivityVariableType variableType,
-                              String variableId, boolean variableSet, ContingencyContext contingencyContext) {
+                              String variableId, Boolean variableSet, ContingencyContext contingencyContext) {
```

Comment:
I think it would be cleaner to keep the parameter's type to `boolean`. It defines a "contract": this parameter should not be null.
The problem here is that this method is called with an unchecked `Boolean` variable that will be unboxed. So it would be better to fix the caller which does not respect this contract.
Therefore, can you do the following?
- Keep the `boolean` type for this variable;
- Change `SensitivityFactor.parseJson(JsonParser parser)` to call the constructor with `Objects.requireNonNull(context.variableSet)` instead of `context.variableSet`.